tls/s2n_server_cert.c

```diff
@@ -37,33 +37,18 @@ int s2n_server_cert_recv(struct s2n_connection *conn)
         S2N_ERROR(S2N_ERR_BAD_MESSAGE);
     }
 
-    int certificate_count = 0;
-    while (s2n_stuffer_data_available(&conn->handshake.io)) {
-        uint32_t certificate_size;
+    struct s2n_cert_public_key public_key;
+    struct s2n_blob cert_chain;
+    cert_chain.data = s2n_stuffer_raw_read(&conn->handshake.io, size_of_all_certificates);
+    cert_chain.size = size_of_all_certificates;
 
-        GUARD(s2n_stuffer_read_uint24(&conn->handshake.io, &certificate_size));
+    GUARD(conn->verify_cert_chain_cb(&cert_chain, &public_key, conn->verify_cert_context));
```

Comment:
That was intentional. I was trying to keep a separation of concerns between `s2n_server_cert_recv` and `verify_cert_chain_cb`. `verify_cert_chain_cb` is going to need to check for proper lengths any and do all of the same validations, I'd rather have those validations all in one place than in two places.